lib/puppet/gettext/config.rb

```diff
@@ -67,14 +78,47 @@ def self.reset_text_domain(domain_name)
   # @return true if Puppet translations were successfully loaded, false
   # otherwise
   def self.create_default_text_domain
-    return unless gettext_loaded?
+    return if @gettext_disabled || !gettext_loaded?
 
-    FastGettext.add_text_domain('puppet', type: :chain, chain: [])
-    FastGettext.default_text_domain = 'puppet'
+    FastGettext.add_text_domain(DEFAULT_TEXT_DOMAIN, type: :chain, chain: [])
+    FastGettext.default_text_domain = DEFAULT_TEXT_DOMAIN
 
     load_translations('puppet', puppet_locale_path, translation_mode(puppet_locale_path))
   end
 
+  # @api private
+  # Switches the active text domain, if the requested domain exists.
+  # @param [String] domain_name the name of the domain to switch to
+  def self.use_text_domain(domain_name)
+    return if @gettext_disabled || !gettext_loaded?
+
+    if FastGettext.translation_repositories[domain_name]
```

Comment:
Might be more direct to test `FastGettext.translation_repositories.include?(domain_name)` and avoids the problem if the value is falsey (false or nil)